src/Symfony/Component/Finder/Finder.php

```diff
@@ -669,7 +669,8 @@ public function append($iterator)
         } elseif ($iterator instanceof \Traversable || \is_array($iterator)) {
             $it = new \ArrayIterator();
             foreach ($iterator as $file) {
-                $it->append($file instanceof \SplFileInfo ? $file : new \SplFileInfo($file));
+                $file = $file instanceof \SplFileInfo ? $file : new \SplFileInfo($file);
+                $it[$file->getRealPath()] = $file;
```

Comment:
This won't work for non-existing files. Not sure if that's an issue though.

Reply:
Instead of `getRealPath()`, I suggest using`getPathname()`, which would also work for stream wrappers not supporting `realpath()`.

Non-existent files are OK to me. I don't think this is an expected supported case for the Finder anyway.